### PR TITLE
plumbing: format/packfile, remove duplicate checks in findMatch()

### PR DESCRIPTION
**Pull request by edigaryev**
At the beginning of `findMatch()`, there exist two checks for `len(tgt) < tgtOffset+s` and `len(src) < blksz`.

Simplify the code by removing the inverse checks `len(tgt) >= tgtOffset+s` and `len(src) >= blksz` down below.

### CODE REVIEW

File: plumbing/format/packfile/delta_index.go
Comment (onee-only) [inline plumbing/format/packfile/delta_index.go:45]:
```suggestion
	h := hashBlock(tgt, tgtOffset)
	eIdx := idx.table[h & idx.mask]
	if eIdx == 0 {
		return
	} 
	
	srcOffset = idx.entries[eIdx]
	l = matchLength(src, tgt, tgtOffset, srcOffset)
```
I think applying early return could simplify the function a bit more.  How do you think @edigaryev?

Comment (edigaryev) [reply plumbing/format/packfile/delta_index.go]: Sounds good, implemented in 932b08f13352fdcb9c193908bafac83e0dabaac0.

**Review by onee-only — APPROVED** [review]
LGTM 👍 . Unfortunately, I cannot merge this since I'm not the maintainer of this project.

**Review by pjbgf — APPROVED** [review]
LGTM

**Comment by pjbgf** [comment]
Thanks @onee-only for the initial review and @edigaryev for working on this. :bow:

**Pull request merged**
